lib/index.js

```diff
@@ -291,6 +291,10 @@ function serializeField(schema, fieldName, value, fieldType, writer) {
     }
 }
 function serializeStruct(schema, obj, writer) {
+    if (typeof obj.serialize === 'function') {
+        obj.serialize(writer);
```

Comment:
I would prefer obj.borshSerialize/obj.borshDeserialize for a less chance of name collision